django_filters/filterset.py

```diff
@@ -357,6 +362,18 @@ def get_filters(cls):
         filters.update(cls.declared_filters)
         return filters
 
+    @classmethod
+    def handle_unrecognized_field(cls, field_name, message):
+        behavior = cls._meta.unknown_field_behavior
+        if behavior == "raise":
+            raise AssertionError(message)
+        elif behavior == "warn":
+            warnings.warn(f"Unrecognized field type for '{field_name}'. Field will be ignored.")
+        elif behavior == "ignore":
+            pass
+        else:
+            raise ValueError(f"Invalid unknown_field_behavior: {behavior}")
```

Comment:
This check could occur in `FilterSetOptions.__init__()`. We could just assert a known value, raising early at that point. 

Q: Do we want to use an `Enum` rather than string values?

Reply:
Thanks for taking a look at the code! Will update the first point. Regarding your Enum question, that's a good idea. It enforces type safety at least, looks cleaner as well. I'll add it in a new commit as well.

Reply:
Question: do we want users to still define the `unknown_field_behavior` option as `string`, or would they need to import the `Enum` and choose from the available values instead?

Reply:
I added both the `Enum` type as well as the early raise of `ValueError`. Updated the tests accordingly.